### PR TITLE
Add `content` prop

### PR DESCRIPTION
**Pull request by irudoy**
- [x] test
- [x] docs
- [x] story

### CODE REVIEW

File: src/storybook/html.stories.tsx [hunk 1]
```diff
@@ -7,6 +7,41 @@ const storyMeta: Meta = {
   title: 'Demo/Html',
 }
 
+const Content: React.FC<{ text: string }> = ({ text }) => {
+  const [count, setCount] = React.useState(1)
+
+  React.useEffect(() => {
```
Comment (akhmadullin) [inline src/storybook/html.stories.tsx:13]: in other stories we use `useState` and `useEffect` directly - without `React.`

File: src/gallery.tsx [hunk 2]
```diff
@@ -92,6 +100,8 @@ export const Gallery: FC<GalleryProps> = ({
   withCaption,
   withDownloadButton,
 }) => {
+  const [contentPortal, setContentPortal] = useState(null)
```
Comment (akhmadullin) [inline src/gallery.tsx:103]: mb add explicit type definition?
```tsx
const [contentPortal, setContentPortal] = useState<ReactPortal | null>(null)
```

File: src/item.ts [hunk 3]
```diff
@@ -66,7 +66,12 @@ export interface ItemProps {
...
   /**
-   * Html content, if you need to use it as modal
+   * Slide content
+   */
+  content?: JSX.Element
+
+  /**
+   * Slide html content
    */
   html?: string
```
Comment (akhmadullin) [inline src/item.ts:76]: Is it still actual for React wrapper to have html prop if we can provide JSX content?

Although, it still can be useful for some people/cases 🤔

File: src/storybook/html.stories.tsx [hunk 4]
```diff
@@ -7,6 +7,41 @@ const storyMeta: Meta = {
...
+      <h1 style={{ fontWeight: 'normal' }}>
+        {Array.from(
+          { length: count },
+          () => cats[Math.floor(Math.random() * cats.length)],
+        )}
+      </h1>
+      {text}
+    </div>
+  )
+}
```
Comment (akhmadullin) [inline src/storybook/html.stories.tsx:43]: https://dromru.github.io/react-photoswipe-gallery/slide-content/?path=/story/demo-html--html

something went wrong with story source code addon

source code is not valid

File: src/gallery.tsx [hunk 5]
```diff
@@ -143,6 +154,8 @@ export const Gallery: FC<GalleryProps> = ({
           msrc: thumbnail,
           element: ref.current,
           thumbCropped: cropped,
+          content,
+          ...(content !== undefined ? { type: 'html' } : {}),
```
[the remark this answers is not in the document]
Comment (irudoy) [reply src/gallery.tsx:158]: yep, we should give precedence to manually defined `type` slide option

File: src/storybook/html.stories.tsx (hunk 1 above)
Comment (irudoy) [reply src/storybook/html.stories.tsx:13]: thx, missed it while prototyping

File: src/item.ts (hunk 3 above)
Comment (irudoy) [reply src/item.ts:76]: I think we will leave it at this stage

File: src/storybook/html.stories.tsx (hunk 4 above)
Comment (irudoy) [reply src/storybook/html.stories.tsx:43]: It's fine 😆

File: src/gallery.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (irudoy) [reply src/gallery.tsx:158]: it will overwrite it from `rest` now, looks good

**Pull request merged**